freeipa.spec.in

```diff
@@ -473,6 +473,8 @@ Requires: gssproxy >= 0.7.0-2
 Requires: sssd-dbus >= %{sssd_version}
 Requires: libpwquality
 Requires: cracklib-dicts
+# NDR libraries are internal in Samba and change with version without changing SONAME
+Requires: samba-client-libs >= %{samba_version}
```

Comment:
Currently we have the following:
on rhel: samba_version 4.12.3-12
on fedora: samba_version 2:4.12.10

BuildRequires:  samba-devel >= %{samba_version}
and samba-client-libs is required by sssd-ipa/freeipa-client (version is determined by the samba-devel version used for building sssd).

This patch does not add a new dependency, and does not force a dependency on samba-client-libs version >= the one used at build time (for instance we can still build with 4.13 and install 4.12.10). The patch needs to retrieve the samba version with something like `%global samba_package_version %(rpm -q samba-devel --queryformat %{version}-%{release})`

Reply:
We used to have `%requires_eq` in RPM that could be used for that -- we used it in Samba spec. Then it got removed in Fedora and we moved it locally to the spec file. So I guess something like this should be added here, with the >= change.

This is the version from samba spec file: https://src.fedoraproject.org/rpms/samba/blob/rawhide/f/samba.spec#_130
```
%samba_requires_eq()  %(LC_ALL="C" echo '%*' | xargs -r rpm -q --qf 'Requires: %%{name} = %%{epoch}:%%{version}\\n' | sed -e 's/ (none):/ /' -e 's/ 0:/ /' | grep -v "is not")
```

And it is used this way: https://src.fedoraproject.org/rpms/samba/blob/rawhide/f/samba.spec#_519
```
# Force using libldb version to be the same as build version
# Otherwise LDB modules will not be loaded and samba-tool will fail
# See bug 1507420
%samba_requires_eq libldb
```

Reply:
As to the comment that it does not add a new dependency, it actually does ensure samba-client-libs is at least the version we were building against. This is a real fix, it helped to fix the same problem in RHEL. We have autogenerated dependencies to ABI of NDR libraries but that's not enough and could be satisfied by the older version. This change is fixing it.

The only problem I see with 'rpm -q` approach is that it would break when the database format of RPM changes. This was the problem when RPM migrated from db2 to sqlite -- I had encountered this in COPR where build environments for SRPM build were old but ran against a new chroot with new database format. As a result, SRPM build failed.